src/main/java/legends/model/WrittenContent.java

```diff
@@ -113,5 +123,367 @@ public String getAuthoredOn() {
+			}
+		} else if (type.contains("poem")) {
+			if (authorRoll < 10) {
+				result = "Overall, the poetry is very, very bad.";
+			} else if (authorRoll >= 10 && authorRoll < 20) {
+				result = "Overall, the poetry is not awful, but not very good either.";
+			} else if (authorRoll >= 20 && authorRoll < 30) {
+				result = "Overall, the poetry is passable.";
+			} else if (authorRoll >= 30 && authorRoll < 40) {
+				result = "Overall, the poetry is great.";
+			} else if (authorRoll >= 40 && authorRoll < 50) {
+				result = "Overall, the poetry is inspired.";
+			} else if (authorRoll >= 50) {
+				result = "Overall, the poetry is masterful.";
+			}
+		} else if (type.contains("musical composition")) {
+			if (authorRoll < 10) {
+				result = "Overall, the composition is atrocious.";
+			} else if (authorRoll >= 10 && authorRoll < 20) {
+				result = "Overall, the composition is not awful, but not very good either.";
+			} else if (authorRoll >= 20 && authorRoll < 30) {
+				result = "Overall, the composition is passable.";
+			} else if (authorRoll >= 30 && authorRoll < 40) {
+				result = "Overall, the composition is great.";
+			} else if (authorRoll >= 40 && authorRoll < 50) {
+				result = "Overall, the composition is inspired.";
+			} else if (authorRoll >= 50) {
+				result = "Overall, the composition is masterful.";
+			}
+		} else if (type.contains("choreography")) {
+			if (authorRoll < 10) {
+				result = "Overall, the choreography is just terrible.";
+			} else if (authorRoll >= 10 && authorRoll < 20) {
+				result = "Overall, the choreography is not awful, but not very good either.";
+			} else if (authorRoll >= 20 && authorRoll < 30) {
+				result = "Overall, the choreography is passable.";
+			} else if (authorRoll >= 30 && authorRoll < 40) {
+				result = "Overall, the choreography is great.";
+			} else if (authorRoll >= 40 && authorRoll < 50) {
+				result = "Overall, the choreography is inspired.";
+			} else if (authorRoll >= 50) {
+				result = "Overall, the choreography is masterful.";
+			}
+		} else {
+			if (authorRoll < 10) {
+				result = "Overall, the prose is amateurish at best.";
+			} else if (authorRoll >= 10 && authorRoll < 20) {
+				result = "Overall, the prose is not awful, but not very good either.";
+			} else if (authorRoll >= 20 && authorRoll < 30) {
+				result = "Overall, the prose is passable.";
+			} else if (authorRoll >= 30 && authorRoll < 40) {
+				result = "Overall, the prose is great.";
+			} else if (authorRoll >= 40 && authorRoll < 50) {
+				result = "Overall, the prose is spendid.";
+			} else if (authorRoll >= 50) {
+				result = "Overall, the prose is masterful.";
+			}
+		}
+		
+		return result;
+	}
+	
+	public String getStyleString() {
+		if (styles.isEmpty()) {
+			return "";
+		}
+		
+		String result = "The writing ";
+		List<String> styleString = new ArrayList<String>();
+		int authorWeight = 50;
+		
+		for (String style : styles) {
```

Comment:
Beware, legends_plus (from the dfhack export script) provides its own <style> tags that provoke an exception:
without legends_plus:    
		<style>vicious:4</style>
		<style>serious:2</style>
with legends_plus:
		<style>vicious:4</style>
		<style>serious:2</style>
		<style>Vicious</style>
		<style>Serious</style>
The extra tags from legends_plus may have to be ignored for the code to work in all cases ?